metadata/params/params_test.go

```diff
@@ -119,6 +119,41 @@ func TestAppendComponentParams(t *testing.T) {
 }`,
 			Params{"replicas": "5", "name": `"foo-bar"`},
 			`
+{
```

Comment:
I'd suggest that these are getting big enough where they should be moved into their own separate files and loaded as fixtures. The test data makes it difficult to see what test is being run.

Reply:
Also, it might be good to take advantage of go subtests as well. This will allow for more descriptive failures instead of the whole testing failing. e.g:

```go
cases := []struct{
  name string
}{}

for _, case := range cases {
  t.Run(string, func(t *testing.T) {
    // where the magic happens
  })
}
```

Reply:
Thanks for the feedback @bryanl  -- I'll open a ticket to do the restructuring of these tests as they can be applied to a few areas of the codebase. #168